rir/src/compiler/pir/instruction.h

```diff
@@ -995,6 +995,8 @@ class FLIE(Subassign1_1D, 4, Effects::Any()) {
     Value* idx() { return arg(2).val(); }
     void updateType() override final {
         maskEffectsAndTypeOnNonObjects(lhs()->type | rhs()->type);
+        if (!lhs()->type.isA(PirType::num() | RType::str))
+            type = type.orObject();
```

Comment:
ugh, that is not nice. first overshooting and then correcting after the fact, with some partial list of exceptions. please fix it at the source. for example
```
if (! PirType(RType::vector)::isA(lhs()->type))
  maskEffectsAndTypeOnNonObjects ...
```
sounds reasonable to me?